payment/payment.go

```diff
@@ -46,6 +46,12 @@ func FromPayment(payment *Payment) (*Payment, error) {
 	return &Payment{payment.Network, nil, hash160(redeem.Script), nil, redeem, nil}, nil
 }
 
+// FromPayment creates a nested Payment struct from script
+func FromScript(script []byte) (*Payment, error) {
```

Comment:
I would add a `network` argument. If its value is `nil`, it should default to the liquid mainnet network.

Reply:
This in order to not have a nil `Network` in the payment. This field is important because the prefix of an address changes based on the network.

Reply:
We would check this for consistency also in the FromPublicKey factory function too then?

Reply:
I added validation in both methods and renamed input param name from 'network' to 'net',
I think we should avoid naming variables same as imported package name, do you agree with this?